torchcde/interpolation_hermite_cubic_bdiff.py

```diff
@@ -0,0 +1,48 @@
+import torch
+from torchcde.interpolation_linear import linear_interpolation_coeffs
+
+
+def _setup_hermite_cubic_coeffs_w_backward_differences(times, coeffs, derivs, device=None):
+    """Compute backward hermite from linear coeffs."""
+    x_prev = coeffs[..., :-1, :]
+    x_next = coeffs[..., 1:, :]
+    # Let x_0 - x_{-1} = x_1 - x_0
+    derivs_prev = torch.cat((derivs[..., [0], :], derivs[..., :-1, :]), axis=-2)
+    derivs_next = derivs
+    x_diff = x_next - x_prev
+    t_diff = (times[1:] - times[:-1]).unsqueeze(-1)
+    # Coeffs
+    a = x_prev
+    b = derivs_prev
+    c = 3 * (x_diff / t_diff ** 2 - b / t_diff) - (derivs_next - derivs_prev) / t_diff
+    d = (1 / (3 * t_diff ** 2)) * (derivs_next - b) - (2 * c) / (3 * t_diff)
```

Comment:
Given that we multiply by three in a moment anyway, we can remove the threes in the denominators here.
```
three_d = ...
```

Not that it's likely to make a huge difference to speed, it's just about keeping things tidy really.